src/mods/freecam.cpp

```diff
@@ -24,15 +24,55 @@ static S16Vec s_rot = {};
 
 static patch::Tramp<decltype(&mkb::event_camera_tick)> s_event_camera_tick_tramp;
 
+struct MergedStickInputs {
+    s32 rawX;
+    s32 rawY;
+    s32 substickX;
+    s32 substickY;
+    s32 triggerL;
+    s32 triggerR;
+};
+
+static void get_merged_stick_inputs(MergedStickInputs& outInputs) {
+    outInputs = {};
+
+    // Accumulate stick inputs from all controllers since we don't always
+    // know which player is active, like in menus
+    // TODO account for d-pad control setting
+    if (!pad::get_exclusive_mode()) {
+        for (s32 i = 0; i < 4; i++) {
+            mkb::PADStatus& status = mkb::pad_status_groups[i].raw;
+            if (status.err == mkb::PAD_ERR_NONE) {
+                outInputs.rawX += status.stickX;
+                outInputs.rawY += status.stickY;
+                outInputs.substickX += status.substickX;
+                outInputs.substickY += status.substickY;
+                outInputs.triggerL += status.triggerLeft;
+                outInputs.triggerR += status.triggerRight;
+            }
+        }
+
+        outInputs.rawX = CLAMP(outInputs.rawX, -128, 127);
+        outInputs.rawY = CLAMP(outInputs.rawY, -128, 127);
+        outInputs.substickX = CLAMP(outInputs.substickX, -128, 127);
+        outInputs.substickY = CLAMP(outInputs.substickY, -128, 127);
+        outInputs.triggerL = CLAMP(outInputs.triggerL, 0, 255);
+        outInputs.triggerR = CLAMP(outInputs.triggerR, 0, 255);
+    }
+}
```

Comment:
Nice. Think this would be worth commonizing across freecam and input display? Maybe it could go in `pad.cpp`

Reply:
I wasn't too sure what they all had in common while I was first writing this so I was kinda afraid to change too much lol. I'll take a stab at it tomorrow though.

Reply:
It's not too too much in common so don't sweat it if it seems overkill. That being said if you do, having a per-analog-input-type getter might make sense (instead of computing a whole struct each time), to match the rest of the `pad` API